### DISCUSSION TITLE
Used one .p12 certificate for 3 or more apps for exporting apk's. Will it affect App Signing ?

### DISCUSSION THREAD

**Discussion by Gokulv617** [airsdk/Adobe-Runtime-Support#1084] [Q&A] [answered]
AIR 33.1.1.575
Adobe Animate

I used one .p12 certificate for exporting apk's to playstore previously for 3 or more apps and so they will be having same SHA1 fingerprint. Now I want to upload .aab to playstore and I'm having a doubt before attempting it. It asks me to do App signing. Since I have used same certificate for more apps, Will it affect the app signing ? I mean, will all the apps be correctly signed and work without issues even with the same .p12 certificate used ? Please clarify me.

![img_20210817(2)](https://user-images.githubusercontent.com/43432782/129688537-bbdb3088-a69c-473f-9073-c6803815d8b4.png)

**Comment by Gokulv617** [comment]
@ajwfrost any info on this ?

**Comment by marchbold** [comment, marked as answer]
Use the "Export and upload a key from Java keystore" and follow the process to get the PEPK tool then use it like below with your p12:

`
java -jar pepk.jar --keystore=YOUR_CERTIFICATE.p12 --alias=1 --output=playappsigning_encrypted_certificate.zip --encryptionkey=XXXXXXXXX
`

You shouldn't have any problem doing this with multiple apps.

**Reply by ponarangan** [reply]
Where to create encryption key?

**Comment by Gokulv617** [comment]
Thank you for clarifying me! I'll try and get back

**Comment by Gokulv617** [comment]
I successfully generated the playappsigning_encrypted_certificate.zip. But it is not extractable. What should I do next ?

**Reply by marchbold** [reply]
Follow the directions on the console to upload the file. It's encrypted.

**Comment by Gokulv617** [comment]
okay

**Comment by Gokulv617** [comment]
Everything went well as per the instructions. Thanks! Can you please tell what is the minSDKversion to use for .aab format ? is 14 good to go ?

`<uses-sdk android:minSdkVersion="14" android:targetSdkVersion="29" />`

Note: Previously APK uploads were using 14 as android:minSdkVersion

**Reply by marchbold** [reply]
We use 19 as the minimum for most of our apps. We find supporting older devices than that a lot of work.

**Reply by marchbold** [reply]
And you'll want to use 30 as the target ask now.

**Comment by Gokulv617** [comment]
Thank you. I successfully uploaded the .aab format.

**Comment by Gokulv617** [comment]
I also get a error while enabling "KeepAndroidStudioOutput = true" in adt.cfg. 

![img_20210821](https://user-images.githubusercontent.com/43432782/130311180-bd40d1eb-cb19-49b0-bba7-3ca0422ad69c.png)

**Reply by Gokulv617** [reply]
> I also get a error while enabling "KeepAndroidStudioOutput = true" in adt.cfg.
> 
> ![img_20210821](https://user-images.githubusercontent.com/43432782/130311180-bd40d1eb-cb19-49b0-bba7-3ca0422ad69c.png)

This error comes even after using the latest updated ANE's

**Reply by ajwfrost** [reply]
@Gokulv617 can you please try using an updated ADT file that can be downloaded from the below link?

https://transfer.harman.com/message/OMsxzq2AtV1mJOC00oKDFj

Thanks

**Reply by Gokulv617** [reply]
> @Gokulv617 can you please try using an updated ADT file that can be downloaded from the below link?
> 
> https://transfer.harman.com/message/OMsxzq2AtV1mJOC00oKDFj
> 
> Thanks
```
# When creating an Android App Bundle, this flag can be used to output the Android Studio project
# files in a subfolder of your build directory (rather than using a temporary location that gets
# cleaned up on exit).
KeepAndroidStudioOutput=true
```
Yes this works fine. I'm not getting any errors now. Can you please tell where that folder will be created ? There is no new folder in my build directory

**Reply by Gokulv617** [reply]
> @Gokulv617 can you please try using an updated ADT file that can be downloaded from the below link?
> 
> https://transfer.harman.com/message/OMsxzq2AtV1mJOC00oKDFj
> 
> Thanks

is this `adt.jar` file stable ? Can I use this and publish to playstore ? or should I wait for new official release from your side ?

**Reply by ajwfrost** [reply]
Hi @Gokulv617  - sorry, I missed the earlier message.. the `AndroidStudioProject` folder should be created under your local folder, which would normally be the one that contains the application descriptor file. Although, I'm just realising that Flash Builder may be using something different as a working directory and then adding a full path for the descriptor.. if you didn't find it there, please let me know and we'll check!

You can use the resulting package to upload to the play store, the ADT is just the tool that packages things and that build is part of the incremental updates that will all go into the next release..

thanks

**Reply by marchbold** [reply]
I'd be careful assuming the folder adt is executed in is anywhere near the project, I know with IntelliJ it uses absolute paths for everything, something like:

```
/path/to/java -Dapplication.home=/path/to/airsdk
              -jar /path/to/airsdk/lib/adt.jar 
              -package ... 
              /path/to/build/output.app 
              /path/to/src/app.xml 
```

Might actually be set to the SDK directory?  I'd suggest you use the output path, the same as the dsym output etc?
